src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,180 @@
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+/*Functions 
+read encoder 
+set motor to raise lift 
+set motor to lower lift 
+shift into low or high gear 
+read top limit switch 
+read bottom limit switch 
+read other limit switches...? 
+ go to specified height
+
+Commands 
+go to specified height 
+make the robot climb 
+raise lift 
+lower lift 
+*/
+
+package frc.team166.robot.subsystems;
+
+import javax.lang.model.util.ElementScanner6;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.SpeedControllerGroup;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.command.Subsystem;
+import frc.team166.chopshoplib.commands.ActionCommand;
+import frc.team166.chopshoplib.commands.CommandChain;
+import frc.team166.chopshoplib.commands.SubsystemCommand;
+import frc.team166.robot.RobotMap;
+import frc.team166.robot.RobotMap.Encoders;
+import edu.wpi.first.wpilibj.DigitalInput;
+
+public class Lift extends Subsystem {
+    // Defines Limit Switches (Digital imputs)
+    DigitalInput TopLimitSwitchLift = new DigitalInput(RobotMap.DigitalInputs.LIFT_LIMIT_SWITCH_BOTTOM);
+    DigitalInput BottomLimitSwitchLift = new DigitalInput(RobotMap.DigitalInputs.LIFT_LIMIT_SWITCH_TOP);
+    // Defines Encoders 
+    Encoder liftEncoder = new Encoder(RobotMap.Encoders.LIFT_A, RobotMap.Encoders.LIFT_B);
+    // Defines Motors 
+    WPI_TalonSRX m_liftmotor = new WPI_TalonSRX(RobotMap.CAN.liftMotorA);
+    WPI_TalonSRX m_liftmotor2 = new WPI_TalonSRX(RobotMap.CAN.liftMotorB);
+    // Defines the previus motors as one motor 
+    SpeedControllerGroup liftdrive = new SpeedControllerGroup(m_liftmotor, m_liftmotor2);
+    DoubleSolenoid LiftTransmission = new DoubleSolenoid(RobotMap.Solenoids.LIFT_TRANSMISSION_A,
+            RobotMap.Solenoids.LIFT_TRANSMISSION_B);
+
+    //enumerator that will be pulled from for the GoToHeight Command
+    public enum LiftHeight {
+        //will be changed
+        kFloor(0), kSwitch(1), kPortal(2), kIntake(3), kScaleLow(4), kScaleHigh(5), kClimb(6);
+
+        @SuppressWarnings("MemberName")
+        private double value;
+
+        //
+        LiftHeight(double value) {
+            this.value = value;
+        }
+
+        public double get() {
+            return value;
+        }
+    }
+
+    public Lift() {
+        //creates a child for the encoders
+        addChild(liftEncoder);
+        addChild(TopLimitSwitchLift);
+        addChild(BottomLimitSwitchLift);
+        //will be changed
+        liftEncoder.setDistancePerPulse(.1);
```

Comment:
The value that we set as the DistancePerPulse should be a preference so that we can change it from the SmartDashboard as needed

Reply:
This probably doesn't need to be a preference, but it should be a "final private double" variable defined at the top of your class. Once we actually know how this relates to distance we should put the math in the code.

Reply:
Agree. Preferences should be things that do not depend on the mechanical properties of the component. Changing the preference would just lead to an incorrect distance. This should be a final (constant).